daemon/logger/awslogs/cloudwatchlogs_test.go

```diff
@@ -163,50 +175,66 @@ func TestNewAWSLogsClientLogFormatHeaderHandler(t *testing.T) {
 	}
 	for _, tc := range tests {
 		t.Run(tc.logFormat, func(t *testing.T) {
+			ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				logFormatHeaderVal := r.Header.Get("x-amzn-logs-format")
+				assert.Equal(t, tc.expectedHeaderValue, logFormatHeaderVal)
+				fmt.Fprintln(w, "{}")
+			}))
+			defer ts.Close()
+
 			info := logger.Info{
 				Config: map[string]string{
 					regionKey:    "us-east-1",
 					logFormatKey: tc.logFormat,
+					endpointKey:  ts.URL,
 				},
 			}
 
-			client, err := newAWSLogsClient(info)
+			client, err := newAWSLogsClient(
+				context.TODO(),
+				info,
+				config.WithCredentialsProvider(credentials.StaticCredentialsProvider{
+					Value: aws.Credentials{AccessKeyID: "AKID", SecretAccessKey: "SECRET", SessionToken: "SESSION"},
+				}),
+			)
 			assert.NilError(t, err)
 
-			realClient, ok := client.(*cloudwatchlogs.CloudWatchLogs)
-			assert.Check(t, ok, "Could not cast client to cloudwatchlogs.CloudWatchLogs")
-
-			buildHandlerList := realClient.Handlers.Build
-			request := &request.Request{
-				HTTPRequest: &http.Request{
-					Header: http.Header{},
-				},
-			}
-			buildHandlerList.Run(request)
-			logFormatHeaderVal := request.HTTPRequest.Header.Get("x-amzn-logs-format")
-			assert.Equal(t, tc.expectedHeaderValue, logFormatHeaderVal)
+			_, err = client.CreateLogGroup(context.TODO(), &cloudwatchlogs.CreateLogGroupInput{LogGroupName: aws.String("foo")})
+			assert.NilError(t, err)
 		})
 	}
 }
 
 func TestNewAWSLogsClientAWSLogsEndpoint(t *testing.T) {
-	endpoint := "mock-endpoint"
+	called := false
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		called = true
```

Comment:
Saw your suggestion after I implemented `atomic.Value`. Anyway, that'll make it easier to go to `atomic.Bool`.

I was thinking about race conditions. We know the handler must return for the test control-flow to hit that last `assert`. So the `val.Store` must have been called before the `val.Load`, so I think we're good here.

I'd rather avoid channels because they're not that hard to deadlock/hang with.